cli/src/main/scala/com/codacy/analysis/cli/formatter/Sarif.scala

```diff
@@ -182,7 +182,8 @@ private[formatter] class Sarif(val stream: PrintStream, val executionDirectory:
   }
 
   private def generatePrimaryLocationHash(issue: Issue, fileContents: List[String]): String = {
-    val issueLineContents = fileContents(Math.max(0, issue.location.line - 1))
+    val issueLineIndex = Math.max(0, issue.location.line - 1)
+    val issueLineContents = if (fileContents.length > issueLineIndex) fileContents(issueLineIndex) else ""
```

Comment:
Using apply on a `List` it is not the best for typesafety since it can throw exceptions.. And iterating on `List`s like this makes me think that maybe a List isn't the best data structure here.. But this is probably another story.
I would try to make it more typesafe with:
```suggestion
    val issueLineContents = fileContents.drop(issue.location.line - 1).headOption.getOrElse("")
```

Reply:
Double check the math with the numbers here :)